File: src/main/kotlin/info/mdamour/gottaGoFinder/helpers/ReadableUnitHelper.kt
package info.mdamour.gottaGoFinder.helpers

import kotlin.math.pow
import kotlin.math.round

class ReadableUnitHelper(private val size: Long) {
    fun display(): String {
        return when {
            size > 2.0.pow(30.0) -> "${round(size / 2.0.pow(30.0))} GB"
            size > 2.0.pow(20.0) -> "${round(size / 2.0.pow(20.0))} MB"
            size > 2.0.pow(10.0) -> "${round(size / 2.0.pow(10.0))} KB"
            else -> "$size B"
        }
    }
}

File: web/src/helpers/ReadableUnitHelper.js
function readableUnitHelper(size) {
  if(size > Math.pow(2.0, 30.0))
    return `${Math.round(size / Math.pow(2.0, 30.0))} GB`
  else if(size > Math.pow(2.0, 20.0))
    return `${Math.round(size / Math.pow(2.0, 20.0))} MB`
  else if(size > Math.pow(2.0, 10.0))
    return `${Math.round(size / Math.pow(2.0, 10.0))} KB`
  else
    return `${size} B`
}

export default readableUnitHelper;